source-code/sdk-js/src/helpers/ast.test.ts

```diff
@@ -0,0 +1,215 @@
+import { describe, expect, test } from "vitest"
+import { parse } from "acorn"
+import { wrapVariableDeclaration, insertAst } from "./ast.js"
+// @ts-ignore
+import { generate } from "astring"
+import type { ArrowFunctionExpression, Program } from "estree"
+import type { Options } from "acorn"
+
+//@ivanhofer - is ecmaVersion 2020 correct here?
+const acornOptions = {
+	ecmaVersion: 2020,
+	sourceType: "module",
+} as Options
+
+describe("ast - wrapVariableDeclaration", () => {
+	test("Wraps a simple arrow function declaration", async () => {
+		const input = `
+            export const load = () => {
+                return { data: true }
+            }
+        `
+		const expected = `
+            export const load = wrapFn(() => {
+                return { data: true }
+            })
+        `
+		const inputAst = parse(input, acornOptions) as unknown as Program
+		const resultAst = await wrapVariableDeclaration(inputAst, "load", "wrapFn")
+		const expectedAst = parse(expected, acornOptions) as unknown as Program
+		const expectedCode = generate(expectedAst)
+		const resultCode = generate(resultAst)
+		expect(resultCode).toEqual(expectedCode)
+	})
+	test("Return error for nonexistent declarator", async () => {
+		const input = `
+            export const load = () => {
+                return { data: true }
+            }
+        `
+		// @ivanohofer - what do you think of the type casting below?
+		const inputAst = parse(input, acornOptions) as unknown as Program
+		const result = wrapVariableDeclaration(inputAst, "blue", "wrapFn")
+		await expect(() => result).rejects.toThrowError("Couldn't find variable declarator.")
+	})
+	test("Doesn't manipulate inputs", async () => {
+		const input = `
+            export const load = () => {
+                return { data: true }
+            }
+        `
+		const inputAst = parse(input, acornOptions) as unknown as Program
+		const inputAstClone = structuredClone(inputAst)
+		await wrapVariableDeclaration(inputAst, "load", "wrapFn")
+		expect(inputAst).toEqual(inputAstClone)
+	})
+})
+
+describe("ast - insertAst", () => {
+	test("Inserts a simple import statement, before", async () => {
+		const insertion = `
+            import { wrapFn } from 'some/path'
+        `
+		const source = `
+            export const load = wrapFn(() => {
+                return { data: true }
+            })
+        `
+		const expected = `
+            import { wrapFn } from 'some/path'
+
+            export const load = wrapFn(() => {
+            return { data: true }
```

Comment:
what happened to the indentation?

Reply:
Will fix - prettier doesn't act on template literals

Reply:
ok, so the whitespace does not affect tests, right?